darwin/future/data_objects/workflow.py

```diff
@@ -0,0 +1,172 @@
+from datetime import datetime
+from enum import Enum
+from typing import List, Optional
+from uuid import UUID
+
+from pydantic import Field
+
+from darwin.future.data_objects.typing import UnknownType
+from darwin.future.pydantic_base import DefaultDarwin
+
+
+class WFDataset(DefaultDarwin):
+    """
+    A class to manage all the information around a dataset on the darwin platform,
+    including validation
+
+    Attributes
+    ----------
+    id: int
+    name: str
+    instructions: str
+
+    Methods
+    ----------
+    __int__: returns the id of the dataset
+    __str__: returns the name of the dataset
+    __repr__: returns a string representation of the dataset
+    """
+
+    id: int
+    name: str
+    instructions: str = Field(min_length=0)
+
+    def __int__(self) -> int:
+        return self.id
+
+    def __str__(self) -> str:
+        return self.name
+
+
+class WFEdge(DefaultDarwin):
+    """
```

Comment:
`WFEdge` seems largely complete, we may find more to add to it later, but it works as a graph node already.

Was considering the validity of a graph solver function.  E.g. a function that confirms the graph is valid within our parameters, this would be simple (it's similar to linked list traversal), but until we need it, lets not bother.

Reply:
What would constitute an unsolvable system? disconnected stages that can't be visited? That would seem an easy check with recursion and a 'visited' array.

Reply:
It's a vague reference, but for instance a workflow with unconnected nodes, or nodes connected in one infinite loop, two loops, etc - these are essentially a graph, but we have imposed logic on that.

I don't know if it's necessary for us to do anything with it, as the backend may already validate it.

Reply:
Validation exists somewhere, but might only be on the frontend. If you do try to create disconnected graphs it complains (only when you click save though, so I assume it's on the backend). It might not be that relevant to double up validation.